azure/converters/spotinstances_test.go

```diff
@@ -20,13 +20,110 @@ import (
 	"fmt"
 	"testing"
 
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v5"
 	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2021-11-01/compute"
 	. "github.com/onsi/gomega"
 	"k8s.io/apimachinery/pkg/api/resource"
 	"k8s.io/utils/ptr"
 	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
 )
 
+func TestGetSpotVMOptionsV2(t *testing.T) {
+	deletePolicy := infrav1.SpotEvictionPolicyDelete
+	type resultParams struct {
+		vmPriorityTypes       *armcompute.VirtualMachinePriorityTypes
+		vmEvictionPolicyTypes *armcompute.VirtualMachineEvictionPolicyTypes
+		billingProfile        *armcompute.BillingProfile
+	}
+	tests := []struct {
+		name             string
+		spot             *infrav1.SpotVMOptions
+		diffDiskSettings *infrav1.DiffDiskSettings
+		want             resultParams
+	}{
+		{
+			name:             "nil spot",
+			spot:             nil,
+			diffDiskSettings: nil,
+			want: resultParams{
+				vmPriorityTypes:       nil,
+				vmEvictionPolicyTypes: nil,
+				billingProfile:        nil,
+			},
+		},
+		{
+			name: "spot with nil max price",
+			spot: &infrav1.SpotVMOptions{
+				MaxPrice: nil,
+			},
+			diffDiskSettings: nil,
+			want: resultParams{
+				vmPriorityTypes:       ptr.To(armcompute.VirtualMachinePriorityTypesSpot),
+				vmEvictionPolicyTypes: nil,
+				billingProfile:        nil,
+			},
+		},
+		{
+			name: "spot with max price",
+			spot: &infrav1.SpotVMOptions{
+				MaxPrice: func(price string) *resource.Quantity {
+					p := resource.MustParse(price)
+					return &p
+				}("1000"),
+			},
+			diffDiskSettings: nil,
+			want: resultParams{
+				vmPriorityTypes:       ptr.To(armcompute.VirtualMachinePriorityTypesSpot),
+				vmEvictionPolicyTypes: nil,
+				billingProfile: &armcompute.BillingProfile{
+					MaxPrice: ptr.To[float64](1000),
+				},
+			},
+		},
+		{
+			name: "spot with ephemeral disk",
+			spot: &infrav1.SpotVMOptions{
+				MaxPrice: nil,
+			},
+			diffDiskSettings: &infrav1.DiffDiskSettings{
+				Option: string(compute.DiffDiskOptionsLocal),
```

Comment:
```suggestion
				Option: string(armcompute.DiffDiskOptionsLocal),
```

Reply:
Actually I can see where we might technically want the old type here even though they're equivalent.

Reply:
Grrr... I think you're right again. 🤦🏻 

The unit test passes either way, however, so is it worth changing now? We will soon pave over all usage of `compute.` anyway...

Reply:
I'm pretty sure it's not gonna bite us either way, but I guess my gut is leaning slightly in favor of the `armcompute` variant.